src/System.Net.Http/src/uap/System/Net/HttpClientHandler.cs

```
@@ -684,6 +720,81 @@ private static bool InitRTNoCacheSupported()
+            serverChain.ChainPolicy.RevocationFlag = X509RevocationFlag.ExcludeRoot;
+            // Authenticate the remote party: (e.g. when operating in client mode, authenticate the server).
+            serverChain.ChainPolicy.ApplicationPolicy.Add(s_serverAuthOid);
+            if (!serverChain.Build(serverCert))
+            {
+                sslPolicyErrors |= SslPolicyErrors.RemoteCertificateChainErrors;
+            }
+
+            // Determine name-mismatch error from the existing WinRT information since .NET X509Chain.Build does not
+            // return that in the X509Chain.ChainStatus fields.
+            foreach (RTChainValidationResult result in args.ServerCertificateErrors)
+            {
+                if (result == RTChainValidationResult.InvalidName)
+                {
+                    sslPolicyErrors |= SslPolicyErrors.RemoteCertificateNameMismatch;
+                    break;
+                }
+            }
+
+            // Get the .NET HttpRequestMessage we saved in the property bag of the WinRT HttpRequestMessage.
+            HttpRequestMessage request = (HttpRequestMessage)args.RequestMessage.Properties[RequestMessageLookupKey];
+
+            // Call the .NET callback.
+            bool success = false;
+            try
+            {
+                success = _serverCertificateCustomValidationCallback(request, serverCert, serverChain, sslPolicyErrors);
+            }
+            catch (Exception ex)
+            {
+                // Save the exception info. We will return it later via the SendAsync response processing.
+                args.RequestMessage.Properties.Add(
+                    SavedExceptionDispatchInfoLookupKey,
+                    ExceptionDispatchInfo.Capture(ex));
+            }
+            finally
+            {
+                serverChain.Dispose();
+                serverCert.Dispose();
+            }
+
+            if (!success)
+            {
+                args.Reject();
```

>I don't like that the API requires call to Reject instead of one to Accept. Not yours, I know.

Actually, my team implemented that. :)  We had long discussions about Accept vs. Reject. The architects concluded that since we were adding this callback and it was "extra" validation, the default behavior for an empty callback function was to accept. This was because one still has to manually turn on the flags to ignore errors. So, by default, adding a WinRT callback adds checks and doesn't allow errors to occur.

That aside, I will see about refactoring this code to make it more maintainable.